# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/mac/GeckoTextMarker.mm
# Commit: 879f36821e41
# Full Hash: 879f36821e411d4a99399b9f5f4f48feeb7a1c8b
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-07-06 21:50:07
# Description:
#   Bug 1650740 - Don't allow text range retrieval when doc tree is not fully constructed. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D82410
# ==============================================================================

diff -r c69d97dae74d -r 879f36821e41 accessible/mac/GeckoTextMarker.mm
--- a/accessible/mac/GeckoTextMarker.mm	Mon Jul 06 17:21:37 2020 +0000
+++ b/accessible/mac/GeckoTextMarker.mm	Mon Jul 06 17:31:17 2020 +0000
@@ -199,6 +199,7 @@
   // Find the first link that is at or after the start offset.
   for (int32_t i = 0; i < linkCount; i++) {
     AccessibleOrProxy link = LinkAt(aCurrent, i);
+    MOZ_ASSERT(!link.IsNull());
     linkStartOffset = StartOffset(link);
     if (aStartIntlOffset <= linkStartOffset) {
       next = link;